modules/javafx.graphics/src/main/native-glass/mac/GlassApplication.m

```diff
@@ -762,13 +768,22 @@ + (jobject)enterNestedEventLoopWithEnv:(JNIEnv*)env
     shouldKeepRunningNestedLoop = YES;
     // Cannot use [NSDate distantFuture] because the period is big the app could hang in a runloop
     // if the event came before entering the RL
-    while (shouldKeepRunningNestedLoop && [theRL runMode:NSDefaultRunLoopMode
-                                              beforeDate:[NSDate dateWithTimeIntervalSinceNow:0.010]])
-    {
-        NSEvent * event = [app nextEventMatchingMask: 0xFFFFFFFF untilDate:nil inMode:NSDefaultRunLoopMode dequeue:YES];
-
-        if (event != nil) {
-            [app sendEvent: event];
+    while (shouldKeepRunningNestedLoop) {
+        // The app will crash if we exceed an undocumented limit on the number of nested
+        // CFRunLoopRun calls.
+        nestedRunLoopRunCount += 1;
+        BOOL ran = [theRL runMode:NSDefaultRunLoopMode
+                       beforeDate:[NSDate dateWithTimeIntervalSinceNow:0.010]];
+        nestedRunLoopRunCount -= 1;
+        if (ran) {
```

Comment:
I could be wrong reading this code, but doesn't the `nestedRunLoopRunCount` value remain unchanged?  L774 gets always cancelled by L777?

Reply:
never mind, it's entering and exiting the loop there.